File: 0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int n = gas.length;

        int sumGas = Arrays.stream(gas).sum();
        
        int sumCost = Arrays.stream(cost).sum();
        
        if(sumGas < sumCost)
            return -1;
        
        int total  = 0;
        int result = 0;
        
        for(int i = 0; i<n; i++) {
            
            total += gas[i] - cost[i];
            
            if(total < 0) {
                total = 0;
                result = i+1;
            }
            
        }

        return result;
    }
}

File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        total_gas = sum(gas)
        total_cost = sum(cost)

        if total_cost > total_gas:
            return -1
        
        result = 0
        current_total = 0

        for i in range(len(gas)):
            current_total += gas[i] - cost[i]

            if current_total < 0:
                result = i+1
                current_total = 0
        
        return result